lightning/backend/local.go

```diff
@@ -95,19 +98,32 @@ type Range struct {
 // localFileMeta contains some field that is necessary to continue the engine restore/import process.
 // These field should be written to disk when we update chunk checkpoint
 type localFileMeta struct {
-	Ts        uint64 `json:"ts"`
-	Length    int64  `json:"length"`
-	TotalSize int64  `json:"total_size"`
+	Ts uint64 `json:"ts"`
+	// Length is the number of KV pairs stored by the engine.
+	Length int64 `json:"length"`
+	// TotalSize is the total pre-compressed KV byte size stored by engine.
+	TotalSize int64 `json:"total_size"`
 }
 
 type LocalFile struct {
 	localFileMeta
 	db   *pebble.DB
 	Uuid uuid.UUID
+
+	// importSemaphore ensures the engine can only be imported by a single
+	// thread. We use a semaphore instead of a sync.Mutex because we need the
+	// TryAcquire method.
+	importSemaphore *semaphore.Weighted
```

Comment:
is it faster than select chan with default branch 🤔

Reply:
actually we don't need TryAcquire now 😅, so switching back to regular mutex.